quartz-core/src/main/java/org/quartz/impl/jdbcjobstore/JobStoreSupport.java

```diff
@@ -2902,6 +2891,18 @@ public Boolean validate(Connection conn, List<OperableTrigger> result) throws Jo
                     if (rowsUpdated <= 0) {
                         continue; // next trigger
                     }
+
```

Comment:
Should we update the trigger state, if nextFireTime is null?

Reply:
Well it should not be in a state that it will be updated ... but you are right.

Thinking about the problem, when nextFireTime is null already the state certainly has changed as well. I think the best approach would be to check whether nextTrigger still is STATE_WAITING.

What do you think? I would be prefer to create a new selectTriggerInState function than make an additional query with selectTriggerState?